### PR TITLE
add SemVer compliance text

### PR DESCRIPTION
**Pull request by pravi**
#### Purpose

To give confidence to users of this library about API stability.

#### Related GitHub issues

Closes: #1831

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,6 +65,16 @@ And then execute:
 $ bundle
 ```
 
+## Semantic Versioning
+
+This project conforms to [semver](http://semver.org/). As a result of this policy, you can (and should) specify a dependency on this gem using the [Pessimistic Version Constraint](http://guides.rubygems.org/patterns/) with two digits of precision. For example:
+
+spec.add_dependency 'active_model_serializers', '~> 1.0'
```
Comment (bf4) [inline README.md]: correction `~> 0.10.0`

File: README.md [hunk 2]
```diff
@@ -65,6 +65,16 @@ And then execute:
...
+
+This means your project is compatible with licensee 1.0 up until 2.0. You can also set a higher minimum version:
```
Comment (bf4) [inline README.md]: typo: `license/version`.  But more generally, I don't think all this text is necessary as it's included in the link documentation.

Comment (bf4) [inline README.md]: Are there any prominent projects that use semver that devote this much text?

Comment (NullVoxPopuli) [inline README.md]: I think most just say, `This project conforms to [semver]`. And that's it.

File: README.md (hunk 1 above)
Comment (NullVoxPopuli) [inline README.md]: I was thinking OP was planning for the future.

File: README.md (hunk 2 above)
Comment (bf4) [inline README.md]: with version.  and maybe something like


```plain
[![SemVer 2.0](https://img.shields.io/badge/SemVer-2.0-lightgrey.svg)](http://semver.org/)
```

[SemVer 2.0](http://semver.org/)

Comment (NullVoxPopuli) [inline README.md]: Hey, that's super concise. I like that

Comment (pravi) [inline README.md]: Many projects do this, full list here https://wiki.debian.org/Teams/Ruby/UpstreamPledge

File: README.md (hunk 1 above)
Comment (pravi) [inline README.md]: Well, its really meaningfull only for a stable release, as even under SemVer, you are allowed to break compatibility in any 0.x version.

Comment (jgrau) [inline README.md]: I would be confused if the example in the readme listed a version that does not exist. I would suggest using the current most recent version. Once we get to `1.0` I am sure a `grep` would catch the outdated version and update it.

**Pull request closed without merging**